File: jHyuk/week11/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
#include <numeric>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer;
    vector<int> psum(sequence.size() + 1);

    int s = 0, e = 1;
    int min = 1000001;

    psum[0] = 0;

    for (int i = 0; i < sequence.size(); i++) {
        psum[i + 1] = psum[i] + sequence[i];
    }

    while (e < sequence.size() + 1 && s <= e) {
        if (psum[e] - psum[s] < k) {
            e++;
        }
        else if (psum[e] - psum[s] > k) {
            s++;
        }
        else {
            if (min > e - s) {
                answer.clear();
                answer.push_back(s);
                answer.push_back(e - 1);
                min = e - s;
            }
            e++;
        }
    }

    return answer;
}

File: Seonwoong/11주차/연속된 부분수열의 합/연속된 부분수열의 합.py
def solution(sequence, k):
    s,e=0,0
    n=len(sequence)
    answer=[]
    
    for i in range(n):
        while s<k and e<n:
            s+=sequence[e]
            e+=1
        if s==k:
            answer.append([i,e-1])
        s-=sequence[i]
        
    answer.sort(key=lambda x: x[1]-x[0])
    
    return answer[0]
